### PR TITLE
Add embedded omniscidb packages, update to omniscidb 5.7.0dev

### PR DESCRIPTION
**Pull request by pearu**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.
* [x] Squash commits

This PR introduces three new outputs:
- `omniscidb-common` - contains additional files that other output packages use
- `omniscidbe` - omniscidb shared library, cpu and cuda builds
- `pyomniscidbe` - omniscidb embedded in python module, cpu and cuda builds

and the additional files in `omniscidb*{cuda,cpu}` are removed, that is, `omniscidb` package requires `omniscidb-common`. 

~This PR uses omniscidb branch https://github.com/pearu/omniscidb/tree/pearu/conda (https://github.com/omnisci/omniscidb/pull/647) and hence the PR can be landed after the branch changes are included in the official omniscidb release.~ Update: https://github.com/omnisci/omniscidb/pull/647 has been merged to upstream master.

### CODE REVIEW

File: recipe/conda_build_config.yaml
Comment (vnlitvinov) [inline recipe/conda_build_config.yaml:25]: do note that this breaks compatibility with a lot of older packages (or, rather, older packages start to be incompatible with OmniSci after this), what's the reason for changing?

Comment (pearu) [reply recipe/conda_build_config.yaml:25]: omnisci switches to requiring c++17 and `cuda>=11` that also means `gcc>=9`.

I can try removing `cdt_name` to ensure that this change is still relevant.

Comment (pearu) [reply recipe/conda_build_config.yaml]: Confirmed, `cdt_name` is not needed and removed now.

**Comment by vnlitvinov** [comment]
@pearu seems you need to add a checksum to the sources line to satisfy conda linter...

**Comment by pearu** [comment]
> @pearu seems you need to add a checksum to the sources line to satisfy conda linter...

Yes, however, as noted in the issue description, we'll need to wait until https://github.com/omnisci/omniscidb/pull/647 has landed and a new omnisci tar-ball is released before landing this PR. So, it is ok to ignore conda linter failure regarding the checksum as the final checksum is unavailable anyway.

**Comment by Garra1980** [comment]
Since omnisci/omniscidb#647 has been committed - can we continue with this?

**Comment by pearu** [comment]
> Since [omnisci/omniscidb#647](https://github.com/omnisci/omniscidb/pull/647) has been committed - can we continue with this?

AFAIK, not quite yet. I think conda feedstocks require release tar-balls with stable sha value. The github provided tar-ball for omniscidb master branch is not stable by definition. So, I think a new omniscidb release is required to finalize this PR.

**Comment by vnlitvinov** [comment]
Technically one can take a tarball from a given revision... but I guess we better wait till a release is cut 😄

**Comment by pearu** [comment]
Hi all,

This PR is now ready for review.

Notice that omniscidb 5.6.0 was released with some DBE components missing and hence the need for a patch. Hopefully, in the next release, we can remove the patch.

**Comment by vnlitvinov** [comment]
@pearu are you serious in requesting a review of **56 thousands**-lines-long patch?..

**Comment by pearu** [comment]
> @pearu are you serious in requesting a review of **56 thousands**-lines-long patch?..

No, to avoid insanity.

The patch is pretty much a compromise of having omniscidb 5.6.0 conda package with DBE support or not having DBE support at all in 5.6.0. The patch can likely be removed in the next release.

**Comment by pearu** [comment]
This PR is ready for review. The upstream has tagged 5.7.0dev that resolves the large patch size issue.

**Comment by vnlitvinov** [comment]
image
@pearu am I looking at the wrong place, or did you forget to update the PR?.. judjing by recent commits you forgot to _remove_ the patch :)

**Comment by pearu** [comment]
> @pearu am I looking at the wrong place, or did you forget to update the PR?.. judjing by recent commits you forgot to _remove_ the patch :)

Ah, right, I have removed using the patch (from meta.yaml) but did not remove the patch file itself. I'll remove it..

File: recipe/meta.yaml
Comment (vnlitvinov) [inline recipe/meta.yaml:1]: maybe you should include `dev` in version?.. to be able to distinguish this and real `5.7.0` releases

File: recipe/meta.yaml
Comment (vnlitvinov) [inline recipe/meta.yaml:13]: this is for 5.7, so either comment has to be updated or constraint lifted :)

File: recipe/meta.yaml
Comment (vnlitvinov) [inline recipe/meta.yaml:243]: why do we require `pytest` in `host`? I thought conda had ways of specifying test-only requirements, and I see you're using it down below

File: recipe/build-omniscidb.sh
Comment (pearu) [reply recipe/build-omniscidb.sh:206]: In short, the `+x` pattern is used to test if an environment variable has been set or not. For more information, see for instance https://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html#tag_02_06_02

File: recipe/meta.yaml
Comment (pearu) [reply recipe/meta.yaml:13]: LLVM 12 is not tested, it is not even landed in conda-forge. So I'll just update the comment.

File: recipe/meta.yaml
Comment (pearu) [reply recipe/meta.yaml:243]: Indeed, for pyomniscidb, pytest in host is not required. Removed.

File: recipe/meta.yaml
Comment (pearu) [reply recipe/meta.yaml:1]: Including `dev` in the version will complicate the specifications because `version` is used in file paths that do not include the `dev` part. I think it is not worth it as `dev` will be gone in a month or so anyway.

File: .ci_support/linux_64_cuda_compiler_versionNone.yaml
Comment (pearu) [reply .ci_support/linux_64_cuda_compiler_versionNone.yaml:26]: When using docker image `quay.io/condaforge/linux-anvil-comp7`, there are dependency conflicts when running pyomnisidb tests:

<details>

```
Unsatisfiable dependencies for platform linux-64: {"python[version='3.7.*|3.8.*|<4.0|>=3.9,<3.10.0a0']", "packaging[version='>=20.0']", "python[version='2.7.*|3.5.*|3.6.*|3.9.*|>=2.7|>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0|>=3.8,<3.9.0a0|>=3.5,<3.6.0a0|3.4.*']", "python[version='2.7.*|3.5.*|3.6.*|3.4.*|>=3.7,<3.8.0a0|>=3.6,<3.7.0a0|>=2.7,<2.8.0a0|>=3.5,<3.6.0a0|>=3.8,<3.9.0a0|>=3.9,<3.10.0a0']", "rbc[version='>=0.2.2']", "pypy3.7[version='>=7.3.3']", "python[version='2.7.*|3.5.*|3.6.*|>=2.7,<2.8.0a0|>=3.9,<3.10.0a0|>=3.5,<3.6.0a0|3.4.*|>=3.6|3.8.*|3.7.*']", "pypy3.7[version='7.3.*|7.3.3.*|7.3.4.*']", "pypy3.6[version='>=7.3.2|>=7.3.3']", "numpy[version='>=1.16,<2.0a0']", "bzip2[version='>=1.0.6,<2.0a0|>=1.0.8,<2.0a0']", 'certifi', "python_abi[version='2.7.*|3.6.*|3.6|3.9.*|3.7.*|3.8.*|3.7',build='*_cp36m|*_pypy36_pp73|*_cp39|*_cp37m|*_cp38|*_pypy37_pp73|*_cp27mu']", 'python=3.9', "python[version='>=3.6,<3.7.0a0|>=3.8,<3.9.0a0|>=3.7,<3.8.0a0']", 'tbb=2021', "certifi[version='>=2016.09|>=2016.9.26']", "python_abi[version='3.6.*|3.8.*|3.7.*',build='*_cp36m|*_cp38|*_cp37m']", "python[version='>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0|>=3.9,<3.10.0a0|>=3.8,<3.9.0a0']", "sqlalchemy[version='>=1.3']", 'thrift=0.13', "python[version='2.7.*|3.5.*|3.6.*|>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.8,<3.9.0a0|>=3.7,<3.8.0a0|>=3.5,<3.6.0a0|3.4.*']", "_openmp_mutex[version='>=4.5']", 'python_abi=2.7[build=*_cp27mu]', "python_abi[version='2.7.*|3.6.*|3.6|3.8.*|3.7.*',build='*_cp36m|*_cp37m|*_pypy36_pp73|*_cp38|*_cp27mu']", "python[version='2.7.*|3.5.*|3.6.*|3.7.*|3.8.*|3.9.*|>=3.5,<3.6.0a0']", "python[version='2.7.*|3.5.*|3.6.*|>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0|>=3.9,<3.10.0a0|>=3.8,<3.9.0a0|>=3.5,<3.6.0a0|3.4.*']", 'tzdata', "certifi[version='>=2017.4.17']", 'six', 'ca-certificates', "python_abi[version='3.6|3.7',build='*_pypy36_pp73|*_pypy37_pp73']", "pypy3.7[version='>=7.3.3|>=7.3.4']", "pypy3.6[version='7.3.*|7.3.0.*|7.3.1.*|7.3.2.*|7.3.3.*']", "python_abi[version='3.7|3.9.*',build='*_pypy37_pp73|*_cp39']", "python[version='2.7.*|3.5.*|3.6.*|>=3.5,<3.6.0a0|>=3.6|>=3|3.9.*|3.7.*|3.8.*']", 'setuptools', "numpy-base[version='1.16.0|1.16.0|1.16.0|1.16.0|1.16.0|1.16.0|1.16.0|1.16.0|1.16.0|1.16.0|1.16.0|1.16.0|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.2|1.16.2|1.16.2|1.16.2|1.16.2|1.16.2|1.16.3|1.16.3|1.16.3|1.16.3|1.16.3|1.16.3|1.16.4|1.16.4|1.16.4|1.16.4|1.16.4|1.16.4|1.16.5|1.16.5|1.16.5|1.16.5|1.16.5|1.16.5|1.16.6|1.16.6|1.16.6|1.16.6|1.16.6|1.16.6|1.16.6|1.16.6|1.16.6|1.16.6|1.16.6|1.16.6|1.16.6|1.16.6|1.16.6|1.16.6|1.16.6|1.16.6|1.17.2.*|1.17.3.*|1.17.4.*|1.18.1.*|1.18.5.*|1.19.1|1.19.1|1.19.1|1.19.1|1.19.1|1.19.1|1.19.2|1.19.2|1.19.2|1.19.2|1.19.2|1.19.2|1.19.2|1.19.2|1.20.1|1.17.0|1.17.0|1.17.0|1.17.0',build='py37h2f8d375_0|py36hde5b4d6_0|py37h2f8d375_0|py37hde5b4d6_0|py36h2f8d375_0|py27h2f8d375_1|py37hde5b4d6_1|py27hde5b4d6_1|py36h2f8d375_0|py27h2f8d375_1|py27hde5b4d6_1|py36hde5b4d6_0|py37h2f8d375_0|py36h2f8d375_0|py37hde5b4d6_0|py27hde5b4d6_0|py27h2f8d375_0|py36h2f8d375_0|py37hde5b4d6_0|py36hde5b4d6_0|py36h2f8d375_0|py37hde5b4d6_0|py38hde5b4d6_0|py38h41b4c56_3|py37h41b4c56_3|py36hdc34a94_3|py37hdc34a94_3|py38hfa32c7d_0|py37h75fe3a5_0|py37h75fe3a5_0|py38hfa32c7d_0|py37h34387ca_0|py37h7d8b39e_0|py38h34387ca_0|py39h7d8b39e_0|py38h7d8b39e_0|py39h34387ca_0|py39h0f7b65f_0|py39h2ae0177_0|py36hfa32c7d_0|py36h75fe3a5_0|py38h75fe3a5_0|py37hfa32c7d_0|py36h75fe3a5_0|py36hfa32c7d_0|py38h75fe3a5_0|py37hfa32c7d_0|py39hdc34a94_3|py38hdc34a94_3|py36h41b4c56_3|py39h41b4c56_3|py39h76555f2_1|py39hfb011de_1|py27hde5b4d6_0|py27h2f8d375_0|py38h2f8d375_0|py36hde5b4d6_0|py36h2f8d375_0|py37h2f8d375_0|py36hde5b4d6_0|py37hde5b4d6_0|py27hde5b4d6_0|py37h2f8d375_0|py27h2f8d375_0|py27hde5b4d6_0|py37h2f8d375_0|py36hde5b4d6_0|py27h2f8d375_0|py27hde5b4d6_0|py37hde5b4d6_0|py27h2f8d375_0|py36h2f8d375_0|py37h2f8d375_0|py36hde5b4d6_1|py37hde5b4d6_1|py36h2f8d375_1|py37h2f8d375_1|py27hde5b4d6_0|py37hde5b4d6_0|py36hde5b4d6_0|py27h2f8d375_0|py37h2f8d375_0|py36hde5b4d6_1|py36h2f8d375_1|py37h2f8d375_1|py36hde5b4d6_0|py27hde5b4d6_0|py27h2f8d375_0|py37hde5b4d6_0|py36h2f8d375_0']", "python[version='>=3.9,<3.10.0a0']", "certifi[version='>=2016.09|>=2016.9.26|>=2017.4.17']", "six[version='>=1.7.2']", "python[version='>=2.7']", "python[version='>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.9,<3.10.0a0|>=3.7,<3.8.0a0|>=3.8,<3.9.0a0']", "_openmp_mutex[version='*|>=4.5',build=*_llvm]", "python[version='3.7.*|3.9.*|3.8.*']", "python_abi[version='2.7.*|3.6|3.7|3.9.*',build='*_pypy36_pp73|*_pypy37_pp73|*_cp39|*_cp27mu']", "pypy3.7[version='7.3.3.*|7.3.4.*|>=7.3.3']", "python_abi[version='2.7.*|3.6.*|3.6|3.7.*|3.7|3.8.*',build='*_cp36m|*_pypy36_pp73|*_pypy37_pp73|*_cp38|*_cp37m|*_cp27mu']", "tbb[version='>=2019.0|>=2019.1|>=2019.4|>=2019.8|>=2020.0|>=2020.2|>=2020.3']", 'requests', "expat[version='>=2.2.9,<3.0.0a0|>=2.3.0,<3.0a0']", "python_abi[version='3.6.*|3.6|3.7.*|3.7|3.9.*|3.8.*',build='*_cp36m|*_cp37m|*_pypy37_pp73|*_cp39|*_cp38|*_pypy36_pp73']", "python_abi[version='3.6|3.6.*|3.7|3.7.*|3.9.*|3.8.*',build='*_cp36m|*_cp37m|*_pypy37_pp73|*_cp39|*_cp38|*_pypy36_pp73']", "pypy3.7[version='7.3.3.*|7.3.4.*']", 'python_abi=3.9[build=*_cp39]', "numpy[version='1.13.*|1.15.*|>=1.14.6,<1.18.0|>=1.14.6,<2.0a0|>=1.16.5,<2.0a0|>=1.16.6,<2.0a0|>=1.17.5,<2.0a0|>=1.19.5,<2.0a0|>=1.9.3,<2.0a0|>=1.15.4,<2.0a0|>=1.11|>=1.11.3,<2.0a0|>=1.12,<1.13.0a0|>=1.14,<1.15.0a0|>=1.13,<1.14.0a0|>=1.10,<1.11.0a0|>=1.11,<1.12.0a0']", "numpy-base[version='>=1.0.14,<2.0a0|>=1.0.6,<2.0a0|>=1.0.2,<2.0a0|>=1.0.4,<2.0a0']"}
```

</details>

File: recipe/conda_build_config.yaml
Comment (pearu) [reply recipe/conda_build_config.yaml:24]: See above, running pyomniscidb tests has dependency conflicts when using quay.io/condaforge/linux-anvil-comp7 image

File: .ci_support/linux_64_cuda_compiler_versionNone.yaml
Comment (pearu) [reply .ci_support/linux_64_cuda_compiler_versionNone.yaml]: OK, it looks like quay.io/condaforge/linux-anvil-comp7 can be still used. The dependency conflicts were due to not having tbb4py available for py 3.9.

File: recipe/conda_build_config.yaml
Comment (pearu) [reply recipe/conda_build_config.yaml]: Dependency conflicts are now resolved.

File: .ci_support/linux_64_cuda_compiler_versionNone.yaml
Comment (pearu) [reply .ci_support/linux_64_cuda_compiler_versionNone.yaml]: For the record, quay.io/condaforge/linux-anvil is not usable due to:
```
ERROR conda.core.link:_execute(502): An error occurred while installing package 'conda-forge::click-7.1.2-pyh9f0ad1d_0'.
FileNotFoundError(2, "No such file or directory: '/opt/conda/bin/python3.6'")
Attempting to roll back.
```

**Comment by pearu** [comment]
@vnlitvinov @jclay, please review.

File: recipe/build-omniscidb.sh
Comment (vnlitvinov) [reply recipe/build-omniscidb.sh:206]: I never before gave a thought that "unset" and "set but null" are different conditions - most shells treat those two as equal :) Thanks for educating me.

File: recipe/meta.yaml
Comment (vnlitvinov) [reply recipe/meta.yaml:1]: But then again, how would you tell apart `5.7.0` built from `5.7.0-dev` tag and `5.7.0` built from `5.7.0` release?..

File: .ci_support/linux_64_cuda_compiler_versionNone.yaml
Comment (vnlitvinov) [reply .ci_support/linux_64_cuda_compiler_versionNone.yaml]: This warrants a more thorough investigation, I don't think we should have such a strict dependency on `click` (which is a Python library to make CLI apps) that should prevent us from keeping compat with most conda-forge packages.

cc @igalink

BTW @pearu I don't see you using `cos7` anymore, seems that you can have already solved that?

File: recipe/conda_build_config.yaml
Comment (vnlitvinov) [inline recipe/conda_build_config.yaml:24]: I wonder what this change really does - what's the difference between the two images?

**Review by vnlitvinov — APPROVED** [review]
LGTM. There's one question for my education :)

File: recipe/meta.yaml
Comment (pearu) [reply recipe/meta.yaml:1]: I'll ask upstream if the next release could be `5.7.1` but we can use `build` number to distinguish between 5.7.0-dev and 5.7.0.

File: recipe/conda_build_config.yaml
Comment (pearu) [reply recipe/conda_build_config.yaml:24]: According to
https://quay.io/repository/condaforge/linux-anvil-comp7/manifest/sha256:93c7a11c8580cc709a154fb7bb4723dea9f466bfc0859df533560317a6d61794
https://quay.io/repository/condaforge/linux-anvil-cos7-x86_64/manifest/sha256:aec245b91a298c9ed32b73d1438ad79e7f71906a94e71b0efef2ff691685a3f2
linux-anvil-comp7 contains Centos 6 and linux-anvil-cos7-x86_64 contains Centos 7.

File: .ci_support/linux_64_cuda_compiler_versionNone.yaml
Comment (pearu) [reply .ci_support/linux_64_cuda_compiler_versionNone.yaml]: re cos7, the explanation is in https://github.com/conda-forge/omniscidb-feedstock/pull/39#discussion_r625920383

File: recipe/meta.yaml
Comment (pearu) [reply recipe/meta.yaml:1]: Confirmed: the next release will be 5.7.0.

Let me try to introduce `base_version=5.7.0` and have `version=5.7.0dev-20210429` as tagged by the upstream..

File: recipe/meta.yaml
Comment (vnlitvinov) [inline recipe/meta.yaml:1]: I only wonder how conda version comparison works... as ideally we'd like to have `5.7.0 > 5.7.0.dev`

[conda docs](https://conda.io/projects/conda/en/latest/user-guide/concepts/pkg-specs.html) say that `conda` uses [PEP 440](https://www.python.org/dev/peps/pep-0440/), which apparently honors this `.devN` scheme, so we're good here.

**Review by vnlitvinov — APPROVED** [review]
LGTM. Good job!

File: recipe/meta.yaml
Comment (pearu) [reply recipe/meta.yaml:1]: Interesting, I found PEP 386 in https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#package-version .

Comment (pearu) [reply recipe/meta.yaml:1]: PEP 440 says "Developmental releases are ordered by their numerical component, immediately before the corresponding release (and before any pre-releases with the same release segment), and following any previous release (including any post-releases)."

File: recipe/meta.yaml
Comment (Garra1980) [inline recipe/meta.yaml:360]: @pearu any reason to avoid python 3.9?

Comment (pearu) [reply recipe/meta.yaml:360]: yes, tbb4py conda package is not available for python 3.9 yet, see https://github.com/conda-forge/tbb-feedstock

Comment (Garra1980) [reply recipe/meta.yaml:360]: Ah, right, thanks!

**Comment by Garra1980** [comment]
Finally, 5.7 released so waiting it in conda-forge ;)

**Comment by pearu** [comment]
@Garra1980 5.7 in conda-forge is WIP: https://github.com/conda-forge/omniscidb-feedstock/pull/49

**Pull request merged**
